mllib/src/main/scala/org/apache/spark/ml/reduction/OneVsRestClassifier.scala

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.reduction
+
+
+import scala.language.existentials
+
+import org.apache.spark.annotation.{AlphaComponent, DeveloperApi}
+import org.apache.spark.ml.{Estimator, Model}
+import org.apache.spark.ml.attribute.BinaryAttribute
+import org.apache.spark.ml.classification.{ClassificationModel, Classifier, ClassifierParams}
+import org.apache.spark.ml.param.Param
+import org.apache.spark.ml.util.{MetadataUtils, SchemaUtils}
+import org.apache.spark.mllib.linalg._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+import org.apache.spark.storage.StorageLevel
+
+/**
+ * Params for [[OneVsRestClassifier]].
+ */
+private[ml] trait OneVsRestParams extends ClassifierParams {
+
+  type ClassifierType = Classifier[F, E, M] forSome {
+    type F ;
+    type M <: ClassificationModel[F,M];
+    type E <:  Classifier[F, E,M]
+  }
+
+  /**
+   * param for the base classifier that we reduce multiclass classification into.
+   * @group param
+   */
+  val baseClassifier: Param[ClassifierType]  =
```

Comment:
For the class name and param name, maybe we can rename `OneVsRestClassifier` to `OneVsRest` and then rename `baseClassifer` to `classifier`. The code should read fine:

~~~scala
val ovr = new OneVsRest()
  .setClassifier(svm)
~~~

~~~python
ovr = OneVsRest(classifier=svm)
~~~

Btw, we should also mention that `featuresCol` and `labelCol` are ignored in the base classifiers.